src/gt4py/backend/gt_backends.py

```diff
@@ -425,14 +428,21 @@ def visit_Stage(self, node: gt_ir.Stage) -> Dict[str, Any]:
         # Initialize symbols for the generation of references in this stage
         self.stage_symbols = {}
         args = []
+        fields_with_variable_offset = set()
+        for field_ref in gt_ir.filter_nodes_dfs(node, gt_ir.FieldRef):
+            if isinstance(field_ref.offset.get(self.domain.sequential_axis.name, None), gt_ir.Expr):
+                fields_with_variable_offset.add(field_ref.name)
         for accessor in node.accessors:
             self.stage_symbols[accessor.symbol] = accessor
             arg = {"name": accessor.symbol, "access_type": "in", "extent": None}
             if isinstance(accessor, gt_ir.FieldAccessor):
                 arg["access_type"] = (
                     "in" if accessor.intent == gt_definitions.AccessKind.READ else "inout"
                 )
-                arg["extent"] = gt_utils.flatten(accessor.extent)
+                if accessor.symbol not in fields_with_variable_offset:
+                    arg["extent"] = gt_utils.flatten(accessor.extent)
+                else:
+                    arg["extent"] = gt_utils.flatten(accessor.extent[:-1]) + [-1000, 1000]
```

Comment:
The hardcoded -1000 / +1000 extents might warrant a comment, can you explain?

Reply:
It was purely meant to reproduce the work done by @gronerl in the experimental GT_FV3 work: https://github.com/VulcanClimateModeling/GT_FV3/blob/master/fv_mapz/fv3_map_stencils.hpp#L107

I'll add a comment.

Reply:
Good to know, thanks!